### PR TITLE
Add post_result action

### PR DESCRIPTION
**Pull request by manasdk**
* The post_result action is used to post notification results to hubot. This new action
  manages result parsing closer to st2 to hide complexity from hubot.
* The channel rule is modified to use this new action.

Requires https://github.com/StackStorm/st2/pull/1475

### CODE REVIEW

**Comment by Kami** [comment]
Cool, I will look into it and test it with my actions and workflows.

I'm not too happy we don't have any tests for that though. I'm trying to figure out what's the best place to add tests for that, I guess we could add some unit tests inline to the pack (packs/hubot/tests/unit/).

File: packs/hubot/actions/post_result.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+def format_remoterunner_result(result):
+    output = format_possible_failure_result(result)
+    output.update({k: format_localrunner_result(v, do_serialize=False)
+                   for k, v in six.iteritems(result)})
+    return _serialize(output)
+
+
+def format_actionchain_result(result):
+    output = format_possible_failure_result(result)
```
Comment (Kami) [inline packs/hubot/actions/post_result.py]: So it looks like we only format the error here - this means we will never propagate the final result of the action chain, right?

Comment (manasdk) [inline packs/hubot/actions/post_result.py]: Final result is not propgated.

**Comment by manasdk** [comment]
> I guess we could add some unit tests inline to the pack (packs/hubot/tests/unit/).

That seems like a good idea. If we could skip tests during deployment would be awesome but not necessary.

I can write the tests at some point. Maybe after openstack or even during while at the booth.

**Pull request merged**